src/util/serializing/HexObjectEncoding.cpp

```diff
@@ -10,8 +11,8 @@ void HexObjectEncoding::addData(const void* data, int len) {
     char* buffer = static_cast<char*>(g_malloc(len * 2));
 
     for (int i = 0; i < len; i++) {
-        int x = static_cast<uint8_t const*>(data)[i];
-        sprintf(&buffer[i * 2], "%02x", x);
+        uint8_t x = static_cast<uint8_t const*>(data)[i];
+        sprintf(&buffer[i * 2], "%02" PRIx8, x);
     }
 
     g_string_append_len(this->data, buffer, len * 2);
```

Comment:
replace data by `std::vector<std::byte>`

Reply:
That would change the signature of `HexObjectEncoding()`. Who uses `HexObjectEncoding()`? I tried to search for it in Xournal++'s source code, but it only references itself, and it's not used anywhere else in the source code other than the corresponding source and header files. Is it a library used in some system?

Reply:
Currently, no-one is using it. Not sure, if we should just remove that class then.
Since this project isn't a library it is not a problem to switch the signature of classes.
It's more matter of how many work do you have to do, to refactor the code. E.g. changing the signature by a change of private fields, this would only require changes in the corresponding source file. But changing a function requires changes in many places.

Reply:
There's no point in fixing the code for `HexObjectEncoding.cpp` if it's just going to be removed though. It might be older code that used it. Maybe it's actually used for something and it would be a mistake to remove it. If, as you say, it's not needed, I think it should be removed.